L06-annotations/src/main/java/ru/denis_ch/my_test_framework/TestExe.java

```diff
@@ -0,0 +1,105 @@
+package ru.denis_ch.my_test_framework;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.Set;
+
+public class TestExe {
+    private int allCntBefore;
+    private int allCntTest;
+    private int allCntAfter;
+
+    private int errCntBefore;
+    private int errCntTest;
+    private int errCntAfter;
+
+
+    public void do_test(String className) {
+
+        Set<Method> setBefor = new HashSet<>();
+        Set<Method> setTest = new HashSet<>();
+        Set<Method> setAfter = new HashSet<>();
+
+        TestExe testExe = new TestExe();
+
+        Class cl = null;
+        try {
+            cl = Class.forName(className);
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+
+        Method[] method = cl.getMethods();
+
+        for (Method md : method) {
+            if (md.isAnnotationPresent(Before.class)) {
+                setBefor.add(md);
+            } else if (md.isAnnotationPresent(Test.class)) {
+                setTest.add(md);
+            } else if (md.isAnnotationPresent(After.class)) {
+                setAfter.add(md);
+            }
+        }
+
+        System.out.println("methods with annotation @Before " + setBefor.size());
+        System.out.println("methods with annotation @After " + setAfter.size());
+        System.out.println("methods with annotation @Test " + setTest.size());
+
+        for (Method test : setTest) {
+            System.out.println("Checking the method " + test.getName() + "; " + test.getParameterCount());
+
+            Object object = null;
+            try {
+                object = cl.getDeclaredConstructor().newInstance();
+            } catch (InstantiationException e) {
+                throw new RuntimeException(e);
+            } catch (IllegalAccessException e) {
+                throw new RuntimeException(e);
+            } catch (InvocationTargetException e) {
+                throw new RuntimeException(e);
+            } catch (NoSuchMethodException e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
Это можно объединить в один `catch`
```suggestion
} catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
                throw new RuntimeException(e);
            }
```

Можно и просто все исключения ловить:
```suggestion
            } catch (Exception e) {
                throw new RuntimeException(e);
            }
```

Reply:
fixed